packages/fmg-core/test/framework/transitions.js

```diff
@@ -0,0 +1,323 @@
+        });
+
+        it("allows a valid transition", async() => {
+            assert(await validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the balances changes", async () => {
+            toState.resolution = otherResolution;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition from the last accept state", async () => {
+            fromState.stateCount = 1;
+            await assertRevert(validTransition(fromState, toState));
+        });
+    });
+
+    describe("accept -> game", () => {
+        beforeEach(() => {
+            fromState = CountingGame.acceptState({ ...defaults, turnNum: 1, stateCount: 1, gameCounter: 3 });
+            toState = CountingGame.gameState({ ...defaults, turnNum: 2, gameCounter: 4 });
+        });
+
+        it("allows a valid transition", async() => {
+            assert(await validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the turnNum doesn't increment", async () => {
+            toState.turnNum = fromState.turnNum;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects any transition where the channel changes", async () => {
+            toState.channel = otherChannel;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition not from the last accept state", async () => {
+            fromState.stateCount = 0;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the game rules are broken", async() => {
+            toState.gameCounter = 2; // game specifies that counter must increment
+            await assertRevert(validTransition(fromState, toState));
+        });
+    });
+
+    describe("game -> game", () => {
+        beforeEach(() => {
+            fromState = CountingGame.gameState({ ...defaults, turnNum: 1, gameCounter: 3 });
+            toState = CountingGame.gameState({ ...defaults, turnNum: 2, gameCounter: 4 });
+        });
+
+        it("allows a valid transition", async() => {
+            assert(await validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the turnNum doesn't increment", async () => {
+            toState.turnNum = fromState.turnNum;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects any transition where the channel changes", async () => {
+            toState.channel = otherChannel;
+            await assertRevert(validTransition(fromState, toState));
+        });
+
+        it("rejects a transition where the game rules are broken", async() => {
+            toState.gameCounter = 2; // game specifies that counter must increment
+            await assertRevert(validTransition(fromState, toState));
```

Comment:
It might be nice to get mocks working somehow, so that we can mock the game rules to disallow a transition.